### ISSUE TITLE
position embedding only operate in visiable patch

### ISSUE THREAD

**Issue by name555difficult** [Pang-Yatian/Point-MAE#33]
I am learning with your work，I think it is a great job, thanks for your contribution.
Now I am learning your code, I have a simple question in the forward function of `MaskTransformer` class,
```
    def forward(self, neighborhood, center, noaug = False):
        # generate mask: True: masked patch; False: visiable patch
        if self.mask_type == 'rand':
            bool_masked_pos = self._mask_center_rand(center, noaug = noaug) # B G
        else:
            bool_masked_pos = self._mask_center_block(center, noaug = noaug)

        # get the embedding token of each group: shape(B G C)
        group_input_tokens = self.encoder(neighborhood)  #  B G C

        batch_size, seq_len, C = group_input_tokens.size()

        x_vis = group_input_tokens[~bool_masked_pos].reshape(batch_size, -1, C) # visiable patch token
        # add pos embedding
        # mask pos center
        masked_center = center[~bool_masked_pos].reshape(batch_size, -1, 3)
        pos = self.pos_embed(masked_center) # pos embedding only operates in visiable patch
        # pos_embed input: masked center(B, G, 3)
        # pos_embed output: masked center position token(G, G, )

        # transformer
        x_vis = self.blocks(x_vis, pos)
        x_vis = self.norm(x_vis)

        return x_vis, bool_masked_pos
```
the bool_masked_pos: `True: masked patch; False: visiable patch`
so `group_input_tokens[~bool_masked_pos]` extracts the visiable patch token
but in code `center[~bool_masked_pos]` extracts the masked patch center coordinate
but I found that the dim 2 of `x_vis` is `26`, the dim 2 of `masked_center` is `26` too in the process of debugging
**I think `center[~bool_masked_pos]` extracts the visiable patch center coordinate actually**
Can you give me some suggestions. Thank you sincerely

**Comment by Pang-Yatian** [comment]
Hi,

Yes, we drop masked patches in the encoder when doing pre-training. So we only need to get coordinates for visible patches. See the framework figure in our paper to get a better understanding.

**Comment by Pang-Yatian** [comment]
'I think center[~bool_masked_pos] extracts the visible patch center coordinate actually'
Yes, your understanding is correct. But what do you mean by 'but in code center[~bool_masked_pos] extracts the masked patch center coordinate'?

**Comment by name555difficult** [comment]
Oh, that's just beacuse I found that the variable name is `masked_center ` for code ` center[~bool_masked_pos].reshape(batch_size, -1, 3)`, I didn't make myself clear, sorry.😂
